### PR TITLE
Simulating a Simple Epidemic Spreading Process on Graphs with Julia

### LINKED ISSUE #5972: Simulating a Simple Epidemic Spreading Process on Graphs with Julia
#### Proposed title of article
Simulating a Simple Epidemic Spreading Process on Graphs with Julia 

#### Proposed article introduction
Graphs are structures that represent a set of objects and the relations between them. Formally, we say the objects are the vertices or nodes and the relations are the edges or links. We can often see graphs *“appearing”* around us. For example:

- vertices are cities and edges are roads that directly connect two cities;
- vertices are pages on the Web and edges are links that connect two pages;
- vertices are people and edges are the friendships between two individuals.

Those are just some of the applications of graphs — the list is huge.

Thanks to the nature of graphs, that is, the representation of pairwise interactions/relations, they are good structures to model the spreading of an epidemic. We can represent the individuals (people or animals, for example) as the vertices, and the contact/exposure as the edges.

We are going to model the epidemic as follows:

- every vertex will be in one of the following states:
    - S — Susceptible
    - I — Infected
    - R — Recovered
- at each time step:
    - some exposed susceptible vertices (neighbors of the previously infected vertices) are going to be infected
    - the previously infected vertices will recover and be no longer susceptible or infectious

That’s a simple model and it’s actually meant to be! But at the end of the tutorial, you will be able to make your own modifications and add new features or dynamics as you want.

#### Key takeaways 
- A simple real application of Graphs
- Manipulating Graphs in Julia
- Spreading process in Graphs

#### Article quality
I'm using Julia, a programming language that's not often explored in Section and I think this will add value to the Section community. The theme is a current theme and it will be explained in a simplified way but encouraging the reader to add their own features and to experiment with different dynamics. I will use Graphs and show how it can be used in Julia and a real application of the structure.

#### References
- [The Julia Programming Language](https://julialang.org/)
- [Graph (Discrete Mathematics) - Wikipedia](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics))
- [Spreading processes on networks](https://aaronclauset.github.io/courses/5352/csci5352_F21_L9.pdf)

### Templates to use as guides
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by csamuelsm**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5972

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+function update_nodes!(G, I, r)
+		next_infected_nodes = []
+		
+		# rest of function here...
+end
+```
+
+Now we can add the `potential_infected_neighbors` sample to `next_infected_nodes`.
+
+```julia
```
Comment (Jethro-magaji) [inline content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md:310]: Explain every block of code at the end for better understanding, do this for every instance that's necessary.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi @csamuelsm 
Well done for such an amazing article. However, there're a few corrections you need to make based on my reviews.
Please, run your article on tools like Grammarly and ProWritingAid for better editing.
Thanks.

**Comment by csamuelsm** [comment]
@Jethro-magaji revisions done!

File: content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+> Pkg.add("Plots")
+> Pkg.add("GraphPlot")
+> Pkg.add("StatsBase")
+> Pkg.add("Cairo")
+> Pkg.add("Fontconfig")
+> Pkg.add("Compose")
+> exit()
+```
+
+Package installation is done, now let’s open Jupyter-lab:
```
[the remark this answers is not in the document]
Comment (Jethro-magaji) [reply content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md:89]: Although using "We" conveys the idea of learning together. Using the "You" perspective allows the writer to make a connection with his or her audience by focusing on the reader. It goes a step further by creating an interactive literary experience, bringing the reader into the story.

You can adjust this for some instances where "You" can replace "We".

File: content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md [hunk 3]
```diff
@@ -392,6 +410,8 @@ function spreading_plot!(G, r)
 end
```
Comment (Jethro-magaji) [inline content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md:410]: The code above isn't properly formated. Please check other code blocks that need formatting and edit them.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]

**Comment by csamuelsm** [comment]
@Jethro-magaji the code formatting problems are because of my marktext editor, it was not the first time this happened. I will not use it anymore! Anyway, the corrections are done!

**Comment by Jethro-magaji** [comment]
> @Jethro-magaji revisions done!

Hi @csamuelsm 
I just noticed that your images are not properly formatted in the link.
Here's an example of the right formatting style "Simulator-interface"

**Comment by csamuelsm** [comment]
@Jethro-magaji Sorry, it was my first time using images in an article. Thank you for pointing that out! The corrections are done!

**Review by Jethro-magaji — APPROVED** [review]
Hi! @csamuelsm 
Well done for coming up with such an amazing article. Please take note of the corrections I made, to minimize revision for your next article and also read the EngEd [docs](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors).
Thanks.

**Comment by hectorkambow** [comment]
jumping on this

**Comment by hectorkambow** [comment]
@csamuelsm Thank you for your patience (as we have recently seen an influx with PR coming in). 

👍 overall nice work - very detailed and straight forward tutorial 
 
Please review and suggestions/changes made by your peer reviewer and/or myself during a final review. 
 
cc @Jethro-magaji Please do the same 👍

**Comment by Jethro-magaji** [comment]
> @csamuelsm Thank you for your patience (as we have recently seen an influx with PR coming in).
> 
> 👍 overall nice work - very detailed and straight forward tutorial
> 
> Please review and suggestions/changes made by your peer reviewer and/or myself during a final review.
> 
> cc @Jethro-magaji Please do the same 👍

Alright, noted. Thanks.

**Pull request merged**